**CLASSIFICATION: NAIVE BAYES**

*Patricia Pepita - 2602174176*

**Load Library & Data**

In [1]:
import pandas as pd
import numpy as np

In [2]:
df=pd.read_csv('Credit Score Classification Dataset.csv')

**EDA**

In [3]:
df.head()

Age  Gender  Income            Education Marital Status  \
0   25  Female   50000    Bachelor's Degree         Single   
1   30    Male  100000      Master's Degree        Married   
2   35  Female   75000            Doctorate        Married   
3   40    Male  125000  High School Diploma         Single   
4   45  Female  100000    Bachelor's Degree        Married   

   Number of Children Home Ownership Credit Score  
0                   0         Rented         High  
1                   2          Owned         High  
2                   1          Owned         High  
3                   0          Owned         High  
4                   3          Owned         High

In [4]:
df.shape

(164, 8)

In [5]:
df.columns

Index(['Age', 'Gender', 'Income', 'Education', 'Marital Status',
       'Number of Children', 'Home Ownership', 'Credit Score'],
      dtype='object')

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 164 entries, 0 to 163
Data columns (total 8 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   Age                 164 non-null    int64 
 1   Gender              164 non-null    object
 2   Income              164 non-null    int64 
 3   Education           164 non-null    object
 4   Marital Status      164 non-null    object
 5   Number of Children  164 non-null    int64 
 6   Home Ownership      164 non-null    object
 7   Credit Score        164 non-null    object
dtypes: int64(3), object(5)
memory usage: 10.4+ KB


In [8]:
df['Credit Score'].value_counts()

Credit Score
High       113
Average     36
Low         15
Name: count, dtype: int64

In [9]:
df['Education'].value_counts()

Education
Bachelor's Degree      42
Master's Degree        36
Doctorate              31
High School Diploma    30
Associate's Degree     25
Name: count, dtype: int64

In [10]:
df['Marital Status'].value_counts()

Marital Status
Married    87
Single     77
Name: count, dtype: int64

In [11]:
df['Home Ownership'].value_counts()

Home Ownership
Owned     111
Rented     53
Name: count, dtype: int64

- Convert Categorical Values to Numeric Values

In [13]:
# (gender, education, marital status, home ownership, credit score)

df['Gender'].mask(df['Gender'] == 'Male', 0, inplace=True)
df['Gender'].mask(df['Gender'] == 'Female', 1, inplace=True)

df['Education'].mask(df['Education'] == 'High School Diploma', 0, inplace=True)
df['Education'].mask(df['Education'] == "Associate's Degree", 1, inplace=True)
df['Education'].mask(df['Education'] == "Bachelor's Degree", 2, inplace=True)
df['Education'].mask(df['Education'] == "Master's Degree", 3, inplace=True)
df['Education'].mask(df['Education'] == 'Doctorate', 4, inplace=True)

df['Marital Status'].mask(df['Marital Status'] == 'Single', 0, inplace=True)
df['Marital Status'].mask(df['Marital Status'] == 'Married', 1, inplace=True)

df['Home Ownership'].mask(df['Home Ownership'] == 'Rented', 0, inplace=True)
df['Home Ownership'].mask(df['Home Ownership'] == 'Owned', 1, inplace=True)

df['Credit Score'].mask(df['Credit Score'] == 'Low', 0, inplace=True)
df['Credit Score'].mask(df['Credit Score'] == 'Average', 1, inplace=True)
df['Credit Score'].mask(df['Credit Score'] == 'High', 2, inplace=True)

In [14]:
df.head()

Age Gender  Income Education Marital Status  Number of Children  \
0   25      1   50000         2              0                   0   
1   30      0  100000         3              1                   2   
2   35      1   75000         4              1                   1   
3   40      0  125000         0              0                   0   
4   45      1  100000         2              1                   3   

  Home Ownership Credit Score  
0              0            2  
1              1            2  
2              1            2  
3              1            2  
4              1            2

- Convert Data Types (Categorical to Numeric)

In [15]:
df['Gender'] = pd.to_numeric(df['Gender'], errors='coerce')
df['Education'] = pd.to_numeric(df['Education'], errors='coerce')
df['Marital Status'] = pd.to_numeric(df['Marital Status'], errors='coerce')
df['Home Ownership'] = pd.to_numeric(df['Home Ownership'], errors='coerce')
df['Credit Score'] = pd.to_numeric(df['Credit Score'], errors='coerce')

In [16]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 164 entries, 0 to 163
Data columns (total 8 columns):
 #   Column              Non-Null Count  Dtype
---  ------              --------------  -----
 0   Age                 164 non-null    int64
 1   Gender              164 non-null    int64
 2   Income              164 non-null    int64
 3   Education           164 non-null    int64
 4   Marital Status      164 non-null    int64
 5   Number of Children  164 non-null    int64
 6   Home Ownership      164 non-null    int64
 7   Credit Score        164 non-null    int64
dtypes: int64(8)
memory usage: 10.4 KB


**Impute Missing Value & Check Outliers**

In [ ]:
# Already done in classification: decision tree (result will be the same)

**Split Data into Train and Test Data**

In [17]:
input_df=df.drop('Credit Score',axis=1)
output_df=df['Credit Score']

In [18]:
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(input_df, output_df, test_size = 0.2, random_state = 0)

In [19]:
from sklearn.naive_bayes import MultinomialNB
import warnings
warnings.filterwarnings('ignore')

NB_class =MultinomialNB()
NB_class.fit(x_train, y_train)


MultinomialNB()

In [20]:
y_predict=NB_class.predict(x_test)

In [21]:
from sklearn.metrics import classification_report
print('\nClassification Report\n')
print(classification_report(y_test, y_predict, target_names=['Low CS', 'Average CS', 'High CS']))


Classification Report

              precision    recall  f1-score   support

      Low CS       1.00      1.00      1.00         1
  Average CS       1.00      0.60      0.75         5
     High CS       0.93      1.00      0.96        27

    accuracy                           0.94        33
   macro avg       0.98      0.87      0.90        33
weighted avg       0.94      0.94      0.93        33



**Model Evaluation**

Between Low, Average, and High CS, the algorithm successfully predict Low CS and High CS with 100% accuracy and High CS with 93% accuracy. It can be seen as well that the recall score for Average CS is kind of low (0.6), meaning there are many false negatives outcome on the prediction.

**Conclusion**

Accuracy of the model is 94%